### PR TITLE
NOTICK - Introduce testing abstract class to clean up all state machine hooks

### PR DESCRIPTION
**Pull request by kyriathar**
The purpose of this class is to include all state machine hooks clean ups in one place, so that it will be less likely to miss a testing hook clean up, after a test. 

Any testing class populating state machine hooks should be inheriting this class. Any new hooks added in the statemachine, should be cleared in the clean up method of this class.

### CODE REVIEW

File: node/src/test/kotlin/net/corda/node/services/StateMachineCleanUp.kt [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package net.corda.node.services
+
+import net.corda.node.services.statemachine.FlowStateMachineImpl
+import net.corda.node.services.statemachine.SingleThreadedStateMachineManager
+import net.corda.node.services.statemachine.StaffedFlowHospital
+import org.junit.After
+
+abstract class StateMachineCleanUp {
```
Comment (williamvigorr3) [inline node/src/test/kotlin/net/corda/node/services/StateMachineCleanUp.kt]: Personally (and this is a matter of opinion so fell free to ignore it) I think we should call this something more generic like `StateMachineTest`. We can then add more generic shared code in here which is shared between the statemachine tests.

**Review by williamvigorr3 — COMMENTED** [review]
So not all the statemachine tests extend `StateMachineCleanUp` (things like KillFlowTest). Should we use this everywhere? Even if these tests are not using Hooks? Might be safer as someone might accidentally use a Hook.

Another thing is `IntegrationTest` is in ENT (which I think all Integration tests in ent as supposed to extend so we can run the DB tests for things like postgres) maybe this should extend that (In ENT only).

File: node/src/test/kotlin/net/corda/node/services/StateMachineCleanUp.kt (hunk 1 above)
Comment (rick-r3) [reply node/src/test/kotlin/net/corda/node/services/StateMachineCleanUp.kt]: Sounds pretty reasonable to me.

File: node/src/test/kotlin/net/corda/node/services/StateMachineCleanUp.kt [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+    // Empty companion object added so that this class gets included in cordapps created by [enclosedCordapp()].
+    // ([collectEnclosedClasses()] in order to add a class in the jar, that class needs to have at least one enclosed class.)
+    companion object
```
Comment (rick-r3) [inline node/src/test/kotlin/net/corda/node/services/StateMachineCleanUp.kt]: How does this work ?  Does this cause the subclass companion object to use this as a super class, and so even when companion object used in the test for calling `collectEnclosedClasses()` it finds this class ?

File: node/src/test/kotlin/net/corda/node/services/StateMachineCleanUp.kt (hunk 1 above)
Comment (kyriathar) [reply node/src/test/kotlin/net/corda/node/services/StateMachineCleanUp.kt]: Done -renamed to `StateMachineTest`-.

File: node/src/test/kotlin/net/corda/node/services/StateMachineCleanUp.kt (hunk 2 above)
Comment (kyriathar) [reply node/src/test/kotlin/net/corda/node/services/StateMachineCleanUp.kt]: It makes the enclosing class (`StateMachineTest`) to contain a class and therefore it gets added to the cordapp jar (it makes condition in `collectEnclosedClasses` to be true (`enclosedClasses.isNotEmpty() == true`)).

**Comment by rick-r3** [comment]
Actually @williamvigorr3 point about integration tests is probably a good one.  I think this means we need two classes (one in `test` one in `integration-test`), since only integration tests should inherit from `IntegrationTest` and you'll only find that out when you get to ENT.

**Pull request closed without merging**